<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>782</width>
    <height>754</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>File Name 1</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_FileName">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>30</y>
      <width>271</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_browse">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>30</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Browse</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>190</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Thay đổi độ sáng tối</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_c">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>190</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_apply">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>190</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Xử lý ảnh mức xám</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_Apply1">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>190</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Xử lý ảnh màu</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>220</y>
      <width>121</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Thay đổi độ tương phản</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_c1">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>220</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_apply_tp">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>220</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Xử lý ảnh mức xám</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_apply_tp1">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>220</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Xử lý ảnh màu</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>250</y>
      <width>151</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Phân ngưỡng ảnh với ngưỡng</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_n">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>250</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_xuly_pn">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>250</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Xử lý phân ngưỡng</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>80</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>File Name 2</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_Filename1">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>80</y>
      <width>271</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_browse1">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>80</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Browse</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>160</y>
      <width>46</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string>Tỷ lệ</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_ratio">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>160</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_Add">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>160</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Cộng 2 ảnh</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_Sub">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>160</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Trừ 2 ảnh</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_showH">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>120</y>
      <width>171</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Gọi thử hàm TCĐ</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_showH1">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>120</y>
      <width>221</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Hiển thị tổ chức đồ ảnh màu (3 thành phần)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_showH3">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>120</y>
      <width>131</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Hiển thị TCĐ ảnh màu</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>280</y>
      <width>161</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Phân ngưỡng với 2 mức ngưỡng</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_n1">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>280</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="ln_n2">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>280</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>215</x>
      <y>280</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Ngưỡng 1</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>345</x>
      <y>280</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Ngưỡng 2</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_p2n">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>280</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Xử lý</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>310</y>
      <width>171</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Biến đổi tuyến tính tổ chức đồ</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_applyTCD">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>310</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Apply</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>340</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Cân bằng tổ chức đồ</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_ApplyCBTCD">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>340</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Apply</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>370</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Lọc ảnh trung bình</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_LocAnhTB_Gray">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>370</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Lọc ảnh mức xám</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_LocAnhTB_Color">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>370</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Lọc ảnh màu</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>400</y>
      <width>141</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Lọc ảnh trung vị (Median)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_LocAnhMedian_Gray">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>400</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Lọc ảnh mức xám</string>
    </property>
   </widget>
   <widget class="QPushButton" name="bt_LocAnhMedian_Color">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>400</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Lọc ảnh màu</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_Gradient">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>430</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Gradient(X)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>430</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Bộ lọc Gradient</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_GradientY">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>430</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Gradient(Y)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>460</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Bộ lọc Roberts</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_Roberts_X">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>460</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Roberts (X)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_Roberts_Y">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>460</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Roberts (Y)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>500</y>
      <width>71</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Bộ Lọc Sobel</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_Sobel_X">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>490</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Sobel (X)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_Sobel_Y">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>490</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Sobel (Y)</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>782</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
